executor/index_lookup_hash_join.go

```diff
@@ -0,0 +1,449 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"context"
+	"time"
+	"unsafe"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/chunk"
+	"github.com/pingcap/tidb/util/codec"
+	"github.com/pingcap/tidb/util/memory"
+	"github.com/pingcap/tidb/util/ranger"
+	"go.uber.org/atomic"
+	"sync"
+)
+
+//IndexNestedLoopHashJoin is the hash join executor
```

Comment:
Comment should start with white space.

Reply:
Could you add more comment to explain the execution workflow like https://godoc.org/github.com/pingcap/tidb/executor#IndexLookUpJoin